src/java/org/apache/cassandra/metrics/UnweightedCacheMetrics.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.metrics;
+
+import com.codahale.metrics.Gauge;
+import org.apache.cassandra.cache.UnweightedCacheSize;
+
+import static org.apache.cassandra.metrics.CassandraMetricsRegistry.Metrics;
+
+/**
+ * Metrics for an unweighted cache like {@code AuthCache}.
+ */
+public class UnweightedCacheMetrics extends AbstractCacheMetrics
```

Comment:
It took me a while to figure out the difference between weighted and unweighted terms, so I assume the users will have to do the same exercise. At the moment,the `UnweightedCache` fully matches the `AuthCache`, so maybe we should classify these types of caches in this way, wdyt? This should be easier to understand for a user.

Reply:
@Mmuzaf are AuthCaches the only caches ever which will be unweighted? I think the general distinction between weighted and unweighted makes sense. Maybe we could extra document that? Othewise I think it is just ok as it is ...

Reply:
In the future, however, weighted and unweighted may not be sufficient for us as well, so I wouldn't want to introduce a new concept at this point, as the `AuthCache` fully covers all our needs. It also sounds clearer without adding pages of documentation, and we can also change it in the future if a new type arises and we will decide to classify it... :-)

Reply:
I have a hard time to figure out what other metrics we would have so we could not categorize them into weighted or unweighted ones.

Reply:
I tend to agree with @smiklosovic that the general distinction between weighted and unweighted makes sense.

> In the future, however, weighted and unweighted may not be sufficient for us as well, so I wouldn't want to introduce a new concept at this point, as the AuthCache fully covers all our needs.

I don't think any new concept is being introducing here, unweighted caches exist in C* you are just exposing them via the new interface. 

I also don't agree that this requires pages of additional documentation; _some_ docs will be required to direct users to the appropriate metrics, so it seems to me a choice between 

`"For metrics related to auth caches, see the AuthCache metrics 'nameX, nameY, nameZ...'"`
vs
`"For metrics related auth caches, see the UnweightedCache metrics 'nameX, nameY, nameZ...'"`

I don't believe that docs for the metrics themselves don't need to go into great detail about the nature of weighted vs unweighted caches.

That's my $0.02, but I don't feel mega-strongly about it so wouldn't object whichever way you decide.